### ISSUE TITLE
Solaris 11 ips_package resource fails to report error if dependencies can't be met

### ISSUE THREAD

**Issue by scotthain** [chef/chef#5132]
## Description

If you install (for example) the `gcc` package, the expectation at this point would be it would install 5.3.0 - however, if the machine doesn't meet the dependencies, it will continue and not report an error, without actually installing anything.

```
           * ips_package[gcc] action install
             - install version 5.3.0 of package gcc
           * ips_package[gnu-grep] action install
             - install version 2.22 of package gnu-grep
```

The same code that `ips.rb` runs executed manually end up with this:

```
root@virtsol01-kitchen-f88e9f52e0a3-zone:~# pkg info -r  gcc
          Name: developer/gcc
       Summary: GCC
      Category: Development/C (org.opensolaris.category.2008)
                Development/C++ (org.opensolaris.category.2008)
                Development/Fortran (org.opensolaris.category.2008)
                Development/GNU (org.opensolaris.category.2008)
                Development/Objective C (org.opensolaris.category.2008)
         State: Not installed
     Publisher: solaris
       Version: 5.3.0
 Build Release: 5.12
        Branch: 5.12.0.0.0.95.0
Packaging Date: April 20, 2016 04:41:13 PM
          Size: 5.46 kB
          FMRI: pkg://solaris/developer/gcc@5.3.0,5.12-5.12.0.0.0.95.0:20160420T164113Z
root@virtsol01-kitchen-f88e9f52e0a3-zone:~# pkg install gcc@5.3.0
Creating Plan (Running solver): \
pkg install: No solution was found to satisfy constraints

maintained incorporations: None


Plan Creation: dependency error(s) in proposed packages:

  No suitable version of required package pkg://solaris/developer/gcc/gcc-gfortran-53@5.3.0,5.12-5.12.0.0.0.95.0:20160420T162426Z found:
    Reject:  pkg://solaris/developer/gcc/gcc-gfortran-53@5.3.0,5.12-5.12.0.0.0.95.0:20160420T162426Z
    Reason:  All versions matching 'require' dependency pkg:/system/library@0.5.11,5.11-0.175.3.1.0.3.0 are rejected
      Reject:  pkg://solaris/system/library@0.5.11,5.11-0.175.3.1.0.3.0:20150925T164358Z
      Reason:  Global zone has an older version of package: pkg://solaris/system/library@0.5.11,5.11-0.175.2.0.0.42.2:20140624T185750Z
  No suitable version of required package pkg://solaris/developer/gcc/gcc-c-53@5.3.0,5.12-5.12.0.0.0.95.0:20160420T162008Z found:
    Reject:  pkg://solaris/developer/gcc/gcc-c-53@5.3.0,5.12-5.12.0.0.0.95.0:20160420T162008Z
    Reason:  All versions matching 'require' dependency pkg:/system/library@0.5.11,5.11-0.175.3.1.0.3.0 are rejected
      Reject:  pkg://solaris/system/library@0.5.11,5.11-0.175.3.1.0.3.0:20150925T164358Z
      Reason:  Global zone has an older version of package: pkg://solaris/system/library@0.5.11,5.11-0.175.2.0.0.42.2:20140624T185750Z
  No suitable version of required package pkg://solaris/developer/gcc/gcc-c++-53@5.3.0,5.12-5.12.0.0.0.95.0:20160420T161757Z found:
    Reject:  pkg://solaris/developer/gcc/gcc-c++-53@5.3.0,5.12-5.12.0.0.0.95.0:20160420T161757Z
    Reason:  All versions matching 'require' dependency pkg:/system/library@0.5.11,5.11-0.175.3.1.0.3.0 are rejected
      Reject:  pkg://solaris/system/library@0.5.11,5.11-0.175.3.1.0.3.0:20150925T164358Z
      Reason:  Global zone has an older version of package: pkg://solaris/system/library@0.5.11,5.11-0.175.2.0.0.42.2:20140624T185750Z
  No suitable version of required package pkg://solaris/developer/gcc/gcc-gobjc-53@5.3.0,5.12-5.12.0.0.0.95.0:20160420T162623Z found:
    Reject:  pkg://solaris/developer/gcc/gcc-gobjc-53@5.3.0,5.12-5.12.0.0.0.95.0:20160420T162623Z
    Reason:  All versions matching 'require' dependency pkg:/developer/gcc/gcc-c-53 are rejected
      Reject:  pkg://solaris/developer/gcc/gcc-c-53@5.3.0,5.12-5.12.0.0.0.95.0:20160420T162008Z
      Reason:  All versions matching 'require' dependency pkg:/system/library@0.5.11,5.11-0.175.3.1.0.3.0 are rejected
        Reject:  pkg://solaris/system/library@0.5.11,5.11-0.175.3.1.0.3.0:20150925T164358Z
        Reason:  Global zone has an older version of package: pkg://solaris/system/library@0.5.11,5.11-0.175.2.0.0.42.2:20140624T185750Z
  No suitable version of required package pkg://solaris/developer/gcc/gcc-common-53@5.3.0,5.12-5.12.0.0.0.95.0:20160420T162207Z found:
    Reject:  pkg://solaris/developer/gcc/gcc-common-53@5.3.0,5.12-5.12.0.0.0.95.0:20160420T162207Z
    Reason:  All versions matching 'require' dependency pkg:/system/library@0.5.11,5.11-0.175.3.1.0.3.0 are rejected
      Reject:  pkg://solaris/system/library@0.5.11,5.11-0.175.3.1.0.3.0:20150925T164358Z
      Reason:  Global zone has an older version of package: pkg://solaris/system/library@0.5.11,5.11-0.175.2.0.0.42.2:20140624T185750Z
root@virtsol01-kitchen-f88e9f52e0a3-zone:~# echo $?
```

I believe the correct behavior would be to throw an exception in this case.
## Chef Version

12.12.15
## Platform Version

Solaris 11
## Replication Case

Tell us what steps to take to replicate your problem.  See [How to create a Minimal, Complete, and Verifiable example](https://stackoverflow.com/help/mcve)
for information on how to create a good replication case.
## Client Output

The relevant output of the chef-client run or a link to a gist of the entire run, if there is one.

The debug output (chef-client -l debug) may be useful, but please link to a gist, or truncate it.
## Stacktrace

Please include the stacktrace.out output or link to a gist of it, if there is one.
### NOTE: CHEF CLIENT BUGS ONLY

This issue tracker is for the code contained within this repo -- `chef-client`, base `knife` functionality (not
plugins), `chef-apply`, `chef-solo`, `chef-client -z`, etc.
- [Server issues](https://github.com/chef/chef-server/issues/new)
- [ChefDK issues](https://github.com/chef/chef-dk/issues/new)
- Cookbook Issues (see the https://github.com/chef-cookbooks repos or search [Supermarket](https://supermarket.chef.io) or GitHub/Google)

**Comment by scotthain** [comment]
@lamont-granquist @thommay fyi

Also the global zone/dependency stuff is irrelevant to the issue, the `pkg install` command should return 1, which *should* be an exception in this case.

**Comment by scotthain** [comment]
More debug output: https://gist.github.com/scotthain/c2bbde5350ab44850bc623970bb37142

**Comment by btm** [comment]
Fixed by #5773
